packages/standalone/install.ps1

```diff
@@ -0,0 +1,268 @@
+    $name = (New-Guid).ToString("N")
+    New-Item -ItemType Directory -Path (Join-Path $tmp $name)
+}
+
+function Find-Latest-Version {
+    return (Invoke-webrequest -UseBasicParsing -URI "https://typespec.blob.core.windows.net/dist/latest.txt").Content.Trim()
+}
+
+function Get-Download-Url {
+    param(
+      [String] $Version, 
+      [String] $filename
+    )
+    
+    if($Version -eq "latest") {
+        $Version = (Find-Latest-Version)
+    }
+
+    return "https://typespec.blob.core.windows.net/dist/$Version/$($filename)"
+}
+
+
+function Get-Filename {
+    param([String] $target)
+
+    return "tsp-$target.zip"
+}
+
+function Install-tsp {
+  param([string]$Version);
+
+  # if a semver is given, we need to adjust it to this format: tsp-v0.0.0
+  if ($Version -match "^\d+\.\d+\.\d+$") {
+    $Version = "tsp-v$Version"
+  }
+  elseif ($Version -match "^v\d+\.\d+\.\d+$") {
+    $Version = "tsp-$Version"
+  }
+
+  $Arch = "x64"
+
+  $tspRoot = if ($env:tsp_INSTALL) { $env:tsp_INSTALL } else { "${Home}\.tsp" }
+  $tspBin = mkdir -Force "${tspRoot}\bin"
+
+  try {
+    Remove-Item "${tspBin}\tsp.exe" -Force
+  } catch [System.Management.Automation.ItemNotFoundException] {
+    # ignore
+  } catch [System.UnauthorizedAccessException] {
+    $openProcesses = Get-Process -Name tsp | Where-Object { $_.Path -eq "${tspBin}\tsp.exe" }
+    if ($openProcesses.Count -gt 0) {
+      Write-Output "Install Failed - An older installation exists and is open. Please close open tsp processes and try again."
+      return 1
+    }
+    Write-Output "Install Failed - An unknown error occurred while trying to remove the existing installation"
+    Write-Output $_
+    return 1
+  } catch {
+    Write-Output "Install Failed - An unknown error occurred while trying to remove the existing installation"
+    Write-Output $_
+    return 1
+  }
+
+  $target = "tsp-windows-$Arch"
+  $filename = "$target.zip"
+  $URL = GET-Download-Url -Version $Version -Filename $filename
+  $temp = (New-TemporaryDirectory)
+  $ZipPath = "$temp\$filename"
+
+  $DisplayVersion = $(
+    if ($Version -eq "latest") { "tsp" }
+    elseif ($Version -eq "canary") { "tsp Canary" }
+    elseif ($Version -match "^tsp-v\d+\.\d+\.\d+$") { "tsp $($Version.Substring(4))" }
+    else { "tsp tag='${Version}'" }
+  )
+
+  $null = mkdir -Force $tspBin
+  Remove-Item -Force $ZipPath -ErrorAction SilentlyContinue
+
+  # curl.exe is faster than PowerShell 5's 'Invoke-WebRequest'
+  # note: 'curl' is an alias to 'Invoke-WebRequest'. so the exe suffix is required
+  if (-not $DownloadWithoutCurl) {
+    curl.exe "-#SfLo" "$ZipPath" "$URL" 
+  }
+  if ($DownloadWithoutCurl -or ($LASTEXITCODE -ne 0)) {
+    Write-Warning "The command 'curl.exe $URL -o $ZipPath' exited with code ${LASTEXITCODE}`nTrying an alternative download method..."
+    try {
+      # Use Invoke-RestMethod instead of Invoke-WebRequest because Invoke-WebRequest breaks on
+      # some machines, see 
```

Comment:
I'm left in suspense!

Reply:
haha this is a fork of the bun logic where they didn't actually say why